query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs

```diff
@@ -32,13 +32,25 @@ impl ExecutorProcess {
     fn new() -> Result<ExecutorProcess> {
         let (sender, receiver) = mpsc::channel::<ReqImpl>(300);
 
-        std::thread::spawn(|| match start_rpc_thread(receiver) {
+        let handle = std::thread::spawn(|| match start_rpc_thread(receiver) {
             Ok(()) => (),
             Err(err) => {
                 exit_with_message(1, &err.to_string());
             }
         });
 
+        std::thread::spawn(move || {
```

Comment:
Out of curiosity, why did you chose to use another thread and use the join handle, rather than unwinding the panic in the main thread? and exiting?

Reply:
I don't think it would be possible to unwind the panic in the main thread but unwinding in the RPC thread like this
```diff
diff --git a/query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs b/query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs
index 7059d64dc6..df9a106c5c 100644
--- a/query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs
+++ b/query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/js/external_process.rs
@@ -32,10 +32,16 @@ impl ExecutorProcess {
     fn new() -> Result<ExecutorProcess> {
         let (sender, receiver) = mpsc::channel::<ReqImpl>(300);
 
-        std::thread::spawn(|| match start_rpc_thread(receiver) {
-            Ok(()) => (),
-            Err(err) => {
-                exit_with_message(1, &err.to_string());
+        std::thread::spawn(|| {
+            match std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| start_rpc_thread(receiver))) {
+                Ok(Ok(())) => (),
+                Ok(Err(err)) => {
+                    exit_with_message(1, &err.to_string());
+                }
+                Err(err) => exit_with_message(
+                    1,
+                    &format!("panic in rpc thread: {}", err.downcast::<String>().unwrap_or_default()),
+                ),
             }
         });
 ```
is a possible alternative, I didn't actually consider it.